pyproject.toml

```diff
@@ -76,16 +84,23 @@ nmslib-metabrainz = {version = "^2.1.3", python = ">=3.11, <3.13", optional = tr
 
 # The latest torch version available for MacOSX + x86_64 is 2.2.2
 torch = [
-    {version = ">=1.6.0, <2.3.0", markers = "sys_platform == 'darwin' and platform_machine == 'x86_64'", optional = true},
-    {version = ">=1.6.0, <3.0.0", optional = true}
+    {version = ">=1.6.0, <2.3.0", python = "<3.13", markers = "sys_platform == 'darwin' and platform_machine == 'x86_64'", optional = true},
+    {version = ">=1.6.0, <3.0.0", python = "<3.13", optional = true},
+    {version = ">=2.6.0, <3.0.0", python = ">=3.13", optional = true},
+]
+pytorch-lightning = [
+    {version = ">=1.6.0, <3.0.0", python = "<3.13", optional = true},
+    {version = ">=2.5.1, <3.0.0", python = ">=3.13", optional = true},
 ]
-pytorch-lightning = {version = ">=1.6.0, <3.0.0", optional = true}
 
 ipywidgets = {version = ">=7.7,<8.2", optional = true}
 plotly = {version="^5.22.0", optional = true}
 nbformat = {version = ">=4.2.0", optional = true}
-cupy-cuda12x = {version = "^13.3.0", python = "<3.13", optional = true}
-
+cupy-cuda12x = [
+    {version = "^13.3.0", python = "<3.13", optional = true},
+    {version = "^13.4.0", python = ">=3.13", optional = true},
+]
+fastrlock = {version = "^0.8.3", optional = true}
```

Comment:
Could you please add a comment on why we need this? It seems this package will never be installed since it's optional and not included in `extras`

Reply:
fastrlock is a dependency of cupy, and poetry can't resolve the version for Python 3.13 automatically.
To give a hint for poetry, I explicitly gave the version condition.

Here is the commit:
https://github.com/MobileTeleSystems/RecTools/commit/c952fec0a6a57460c6fef1ad8853ec8d96992083

I added a comment for it as well d7d6be7